### PR TITLE
Fix percolator query analysis for function_score query

### PR DESCRIPTION
**Pull request by martijnvg**
* The `verified` flag of the sub result was ignored.
* The `matchAllDocs` flag of the sub result was also ignored. 

Both issues could cause incorrect results.

### CODE REVIEW

**Comment by martijnvg** [comment]
cc: @elastic/es-search-aggs

File: modules/percolator/src/main/java/org/elasticsearch/percolator/QueryAnalyzer.java [hunk 1]
```diff
@@ -461,12 +461,17 @@ static Result analyze(Query query, Version indexVersion) {
...
             // so in that case we set verified to false.
             // (if it matches with the percolator document matches with the extracted terms.
             // Min score filters out docs, which is different than the functions, which just influences the score.)
             boolean verified = functionScoreQuery.getMinScore() == null;
-            return new Result(verified, result.extractions, result.minimumShouldMatch);
+            if (result.matchAllDocs) {
+                return new Result(result.matchAllDocs, verified);
+            } else {
+                return new Result(verified, result.extractions, result.minimumShouldMatch);
+            }
```
Comment (jpountz) [inline modules/percolator/src/main/java/org/elasticsearch/percolator/QueryAnalyzer.java]: I suspect that the fact that it doesn't look at the value of `result.verified` might cause bugs?

**Review by jpountz — CHANGES_REQUESTED** [review]

File: modules/percolator/src/main/java/org/elasticsearch/percolator/QueryAnalyzer.java (hunk 1 above)
Comment (martijnvg) [reply modules/percolator/src/main/java/org/elasticsearch/percolator/QueryAnalyzer.java]: yeah that was wrong the whole time.

Comment (martijnvg) [reply modules/percolator/src/main/java/org/elasticsearch/percolator/QueryAnalyzer.java]: Ok, I have changed that.

**Review by jpountz — APPROVED** [review]
This looks good to me. Can you add a test for the case that the sub query is not verified? (the thing you fixed in the latest commit)

**Comment by martijnvg** [comment]
@jpountz I added such a test in the latest commit.

**Review by jpountz — APPROVED** [review]

**Pull request merged**